ui/desktop/src/components/ApiKeyTester.tsx

```diff
@@ -0,0 +1,232 @@
+import { toastService } from '../toasts';
+import { Key } from './icons/Key';
+import { ArrowRight } from './icons/ArrowRight';
+import { Button } from './ui/button';
+
+interface ApiKeyTesterProps {
+  onSuccess: (provider: string, model: string) => void;
+  onStartTesting?: () => void;
+}
+
+interface TestResult {
+  provider: string;
+  success: boolean;
+  model?: string;
+  totalModels?: number;
+  error?: string;
+  suggestions?: string[];
+}
+
+export default function ApiKeyTester({ onSuccess, onStartTesting }: ApiKeyTesterProps) {
+  const [apiKey, setApiKey] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [testResults, setTestResults] = useState<TestResult[]>([]);
+  const [showResults, setShowResults] = useState(false);
+  const { upsert } = useConfig();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const testApiKey = async () => {
+    const actualValue = inputRef.current?.value || apiKey;
+
+    if (!actualValue.trim()) {
+      toastService.error({
+        title: 'API Key Required',
+        msg: 'Please enter an API key to test.',
+        traceback: '',
+      });
+      return;
+    }
+
+    onStartTesting?.();
+
+    setIsLoading(true);
+    setTestResults([]);
+    setShowResults(true);
+
+    try {
+      const response = await detectCloudProvider({
+        body: { api_key: actualValue },
+        throwOnError: true,
+      });
+
+      if (response.data) {
+        const { provider_name, models } = response.data;
+
+        setTestResults([
+          {
+            provider: provider_name,
+            success: true,
+            model: models[0],
+            totalModels: models.length,
+          },
+        ]);
+
+        const keyName = `${provider_name.toUpperCase()}_API_KEY`;
+        await upsert(keyName, actualValue, true);
+
+        await upsert('GOOSE_PROVIDER', provider_name, false);
+        await upsert('GOOSE_MODEL', models[0], false);
+
+        toastService.success({
+          title: 'Success!',
+          msg: `Configured ${provider_name} with ${models.length} models available`,
+        });
+
+        onSuccess(provider_name, models[0]);
+      }
+    } catch (error: unknown) {
+      setTestResults([
+        {
+          provider: 'Unknown',
+          success: false,
+          error: 'Could not detect provider from API key',
+          suggestions: [
+            'Make sure you are using a valid API key from OpenAI, Anthropic, Google, Groq, or xAI',
+            'Check that the key is complete and not truncated',
+            'Verify your API key is active and has sufficient credits',
+            'For local Ollama setup, use the "Other Providers" section below',
+          ],
+        },
+      ]);
+
+      toastService.error({
+        title: 'Detection Failed',
+        msg: 'Could not detect provider. Please check your API key and try again.',
+        traceback: error instanceof Error ? error.stack || '' : '',
```

Comment:
this doesn't strike me as particularly useful - showing a toast while also displaying the error in the UI. pick on